src/transformers/modeling_utils.py

```diff
@@ -2967,24 +2970,30 @@ def from_pretrained(
                 config._attn_implementation = kwarg_attn_imp
             model_kwargs = kwargs
 
-        # parse model config and init quantizer
-        quantizer = QuantizationConfigParser.get_quantizer(
-            config=config,
-            quantization_config_from_args=quantization_config,
-        )
+        pre_quantized = getattr(config, "quantization_config", None) is not None
+        if pre_quantized or quantization_config is not None:
+            if pre_quantized:
+                config.quantization_config = AutoHFQuantizer.merge_quantization_configs(
+                    config.quantization_config, quantization_config
+                )
+            else:
+                config.quantization_config = quantization_config
+            quantizer = AutoHFQuantizer.from_config(config.quantization_config, pre_quantized=pre_quantized)
```

Comment:
I think it will easier to read + fix if we can call `AutoQuantizationConfig` class methods to make sure that we have indeed a quantization `QuantizationConfigMixin`. Then pass it to `AutoHFQuantizer.from_config` ? With the same behavior as `AutoConfig` and `AutoModel.from_config(AutoConfig)`.